tiledb/api/src/range.rs

```diff
@@ -43,6 +44,37 @@ macro_rules! check_datatype {
     };
 }
 
+fn intersection<'a, B>(
+    left_lower: &'a B,
+    left_upper: &'a B,
+    right_lower: &'a B,
+    right_upper: &'a B,
+) -> Option<(&'a B, &'a B)>
+where
+    B: BitsOrd + ?Sized,
+{
+    if matches!(left_upper.bits_cmp(right_lower), Ordering::Less)
+        || matches!(right_upper.bits_cmp(left_lower), Ordering::Less)
+    {
+        return None;
+    }
+
+    let lower = if matches!(left_lower.bits_cmp(right_lower), Ordering::Less) {
+        right_lower
+    } else {
+        left_lower
+    };
+
+    let upper = if matches!(left_upper.bits_cmp(right_upper), Ordering::Greater)
+    {
+        right_upper
+    } else {
+        left_upper
+    };
+
```

Comment:
Might also be a good idea to assert `lower <= upper` here as well.

Reply:
Done